QEfficient/src/_transformers/auto.py

```diff
@@ -74,11 +100,46 @@ class QEFFAutoModelForCausalLM(QEFFTransformersBase):
     def execute(self, *args, **kwargs): # type: ignore
         raise NotImplementedError("Reached too far!!")
     
-    def export(self):
-        raise NotImplementedError("Reached too far!!")
-    
-    def compile(self, *args, **kwargs) -> Any:
-        raise NotImplementedError("Reached too far!!")
+    def set_runtime(self, runtime: Runtime, runtime_args: Union[QEFFAutoModelForCausalLMAI100RuntimeArgs, QEFFAutoModelForCausalLMCPUORTRuntimeArgs]) -> None:
+        assert runtime != Runtime.CPU_PT, "Please use self.runtime=Runtime.CPU_PT for updating runtime to CPU_PT"
+
+        if runtime == Runtime.CPU_ORT:
+            assert isinstance(runtime_args, QEFFAutoModelForCausalLMCPUORTRuntimeArgs), f"Expected runtime_args of type {QEFFAutoModelForCausalLMCPUORTRuntimeArgs.__class__}, got {type(runtime_args)}"
+            self.ort_runtime_args = runtime_args
+        elif runtime == Runtime.AI_100:
+            assert isinstance(runtime_args, QEFFAutoModelForCausalLMAI100RuntimeArgs),  f"Expected runtime_args of type {QEFFAutoModelForCausalLMAI100RuntimeArgs.__class__}, got {type(runtime_args)}"
+            self.cloud_ai_100_runtime_args = runtime_args
+        self.runtime = runtime
```

Comment:
Why do we need to have only one runtime at a time? We could have something like `execute_pytorch`, `execute_onnxrt` and `execute_aic`. We could still have `self.runtime` set to something as default, that will direct which `execute_` method to be called.

Reply:
Also, we could manage the args internally no? For example, after export, we could directly update the `self.onnx_model_path` instead of needing to set it from outside.

Reply:
If we expose `self.onnx_model_path` as model variable we allow users to set it any value they want, we can pass `forzen=True` in `dataclasses` to not allow such usage.